### DISCUSSION TITLE
How/when does celery create an event loop when using the gevent pool?

### DISCUSSION THREAD

**Discussion by udbhav** [celery/celery#7466] [Q&A]
I'm attempting to send with django-channels in tasks. I assumed that the celery worker in a gevent pool is not using asyncio and isn't running in the asyncio event loop, however an error I get from django channels indicates otherwise: `RuntimeError: You cannot use AsyncToSync in the same thread as an async event loop`. Running a single task works fine, but queuing multiple tasks will result in that error.

Does celery create an event loop? My suspicion is it doesn't happen immediately on startup, but is triggered at some later point under load. Could somebody point me to the relevant code? I tried looking at celery and kombu source and couldn't really figure it out.

Here's a repo to reproduce the issue: https://github.com/udbhav/celery-channels-test
I filed a ticket: https://github.com/celery/celery/issues/7465
Possibly related: https://github.com/django/channels/issues/1799 

Thank you!

**Comment by ashupednekar** [comment]
I'm facing a similar issue too...

In my case, I'm trying to run my task, which consists heavily of async coroutines, over celery gevent pool. I'm not facing any issues while using prefork

**Reply by GuptaAman08** [reply]
Facing the same issue !!. My task calls a coroutine using async.run(). This coroutine has asyncio.gether. The problem comes when messages/tasks are queued and you suddenly start the celery worker with gevent pool.

**Reply by danielWagnerr** [reply]
Same here
